evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -758,3 +762,21 @@ def test_automl_time_series_classification_threshold(mock_binary_fit, mock_binar
         mock_optimize_threshold.assert_not_called()
         assert automl.best_pipeline.threshold == 0.5
         mock_split_data.assert_not_called()
+
+
+@pytest.mark.parametrize("objective", ['F1', 'Log Loss Binary', 'AUC'])
+@patch('evalml.objectives.standard_metrics.F1.optimize_threshold', return_value=0.63)
+@patch('evalml.pipelines.BinaryClassificationPipeline.score')
+@patch('evalml.pipelines.BinaryClassificationPipeline.fit')
+@patch('evalml.pipelines.BinaryClassificationPipeline.predict_proba')
+def test_tuning_threshold_objective(mock_predict, mock_fit, mock_score, mock_get_objective, objective, X_y_binary):
+    X, y = X_y_binary
+    mock_score.return_value = {objective: 0.5}
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type='binary', objective=objective)
+    automl.search()
+
+    if objective != "F1":
```

Comment:
I feel kind of dense here.  What are you testing?  If objective == "F1" that the objective is called more than once?  And if it's "F1" that's only called one time?

Reply:
Ahhh, ok.